### PR TITLE
Rework Getting Started

### PR DESCRIPTION
**Pull request by rylev**

### CODE REVIEW

File: outline/getting-started.md [hunk 1]
```diff
@@ -17,30 +16,24 @@ level tour of necessary knowledge for being productive with Tokio.
...
+**Status**: unassigned.
 
-A crash course into the Future abstraction and how Tokio uses it.
+A high level overview of how and when execution happens in Tokio, how futures
+are lazy and the basics of the polling model.
 
-<a name="tasks"></a>
-## Tasks
+<a name="errors"></a>
+## Error Handling
```
Comment (rylev) [inline outline/getting-started.md]: Not sure if having basics of error handling here works, but might be nice.

File: outline/getting-started.md [hunk 2]
```diff
@@ -17,30 +16,24 @@ level tour of necessary knowledge for being productive with Tokio.
...
 
-**Status**: [Complete](https://tokio.rs/docs/getting-started/tasks/)
+**Status**: unassigned
 
-Crash course into tasks.
+The very basics of error handling
 
 <a name="chat-server"></a>
-## Tasks
+## Example: Chat Server
```
Comment (rylev) [inline outline/getting-started.md]: I still think we need to rethink the example here, but I'll leave that for another PR.

File: outline/getting-started.md (hunk 1 above)
Comment (carllerche) [reply outline/getting-started.md]: Yeah, could work! What would go in the section?

**Comment by carllerche** [comment]
I'm good with this, but would you mind adding a note that it is important to include a visible disclaimer stating that:

1) Tokio's runtime model is significantly different than other languages and to ditch your assumptions. 
2) Tokio's futures are lazy, no action is taken until the future is spawned. If the future is not spawned, no work will ever be done (future is dropped).

Wording will need to be figured out.

File: outline/getting-started.md (hunk 1 above)
Comment (rylev) [reply outline/getting-started.md]: I added some contents down below. I'm not 100% convinced this topic deserves its own section, but I'd need to see it in writing before being able to tell. Thoughts?

Comment (rylev) [reply outline/getting-started.md]: I started playing with implementing this PR based on: https://github.com/tokio-rs/website/pull/223. So far I've simplified the Futures section and moved it front of the Runtime section. With some tweaking it's starting to feel good. I wonder if it's a better way to do things than what I've proposed here.

**Pull request merged**